dataflow/src/backlog/mod.rs

```diff
@@ -231,6 +241,37 @@ impl WriteHandle {
     pub(crate) fn is_partial(&self) -> bool {
         self.partial
     }
+
+    /// Evict `count` randomly selected keys from state and return them along with the number of
+    /// bytes that will be freed once the underlying `evmap` applies the operation.
+    pub fn evict_random_key(&mut self, rng: &mut ThreadRng) -> u64 {
+        let mut bytes_to_be_freed = 0;
+        if self.mem_size > 0 {
+            match self.handle.empty_at_index(rng.gen()) {
+                None => (),
+                Some(vs) => {
+                    let size: u64 = vs.into_iter().map(|r| r.deep_size_of() as u64).sum();
+                    bytes_to_be_freed += size;
```

Comment:
So this doesn't take into account the fact that `evmap` stores the index twice?

Reply:
The rows are deduplicated, and for simplicity we only count memory in rows, not the index entries. Coudl change this, but it wouldn't change any of the fundamentals.